controllers/kubevirtmachine_controller.go

```diff
@@ -568,12 +568,14 @@ func (r *KubevirtMachineReconciler) reconcileKubevirtBootstrapSecret(ctx *contex
 		return errors.New("error retrieving bootstrap data: linked Machine's bootstrap.dataSecretName is nil")
 	}
 
-	// Exit early if exists.
 	bootstrapDataSecret := &corev1.Secret{}
-	bootstrapDataSecretKey := client.ObjectKey{Namespace: vmNamespace, Name: *ctx.Machine.Spec.Bootstrap.DataSecretName + "-userdata"}
-	if err := infraClusterClient.Get(ctx, bootstrapDataSecretKey, bootstrapDataSecret); err == nil {
-		ctx.BootstrapDataSecret = bootstrapDataSecret
-		return nil
+	err := infraClusterClient.Get(
+		ctx,
+		client.ObjectKey{Namespace: vmNamespace, Name: *ctx.Machine.Spec.Bootstrap.DataSecretName + "-userdata"},
+		bootstrapDataSecret)
+	if err != nil && !apierrors.IsNotFound(err) {
+		// not found means not created yet, that's ok. Other errors are something we can't handle.
+		return err
```

Comment:
Is this code really needed now?
There is no other usage for the "bootstrapDataSecret", and anyway the "CreateOrUpdate" func is going to be called

Reply:
good catch. Now that we are always updating the secret, I don't think we need this early exit check if the secret already exists.